core/fetcher.go

```diff
@@ -71,16 +71,24 @@ func (f *BlockFetcher) Commit(ctx context.Context, height *int64) (*types.Commit
 // ValidatorSet queries Core for the ValidatorSet from the
 // block at the given height.
 func (f *BlockFetcher) ValidatorSet(ctx context.Context, height *int64) (*types.ValidatorSet, error) {
-	res, err := f.client.Validators(ctx, height, nil, nil)
-	if err != nil {
-		return nil, err
-	}
+	var perPage = 100
+
+	vals, total := make([]*types.Validator, 0), -1
+	for page := 1; len(vals) != total; page++ {
+		res, err := f.client.Validators(ctx, height, &page, &perPage)
+		if err != nil {
+			return nil, err
+		}
+
+		if res != nil && len(res.Validators) == 0 {
+			return nil, fmt.Errorf("core/fetcher: validators not found")
+		}
 
-	if res != nil && res.Validators == nil {
-		return nil, fmt.Errorf("core/fetcher: validators not found")
+		total = res.Total
+		vals = append(vals, res.Validators...)
 	}
```

Comment:
is this code covered by our current unit tests? If not, can you cover it pls?

Reply:
There is a test calling the ValidatorSet method, but it definitely doesn't fetch more than 100 vals. 
IIRC, there is no way to simulate hundreds of vals in mock client, so not sure how to test this.
Btw, I tested this on my full node and it syncs normally.

Reply:
okay utack

Reply:
I'd be happier somehow having a unit test just for the logic portion.

Reply:
I am happy to make a test if you can show me how. Otherwise, pls, approve.